L1Trigger/L1TTrackMatch/plugins/DisplacedVertexProducer.cc

```diff
@@ -0,0 +1,481 @@
+                                                                         vertex.R_T,
+                                                                         vertex.cos_T,
+                                                                         vertex.delta_z,
+                                                                         vertex.x_dv,
+                                                                         vertex.y_dv,
+                                                                         vertex.z_dv,
+                                                                         vertex.openingAngle,
+                                                                         vertex.p_mag,
+                                                                         isReal);
+
+      std::vector<ap_fixed<13,8,AP_RND_CONV, AP_SAT>> Transformed_features = {selectedTracksEmulationWithTruth[i].first.pt*0.25,
+                                                 selectedTracksEmulationWithTruth[j].first.pt*0.25,
+                                                 selectedTracksEmulationWithTruth[i].first.eta,
+                                                 selectedTracksEmulationWithTruth[j].first.eta,
+                                                 selectedTracksEmulationWithTruth[i].first.phi,
+                                                 selectedTracksEmulationWithTruth[j].first.phi,
+                                                 selectedTracksEmulationWithTruth[i].first.d0,
+                                                 selectedTracksEmulationWithTruth[j].first.d0,
+                                                 selectedTracksEmulationWithTruth[i].first.z0,
+                                                 selectedTracksEmulationWithTruth[j].first.z0,
+                                                 selectedTracksEmulationWithTruth[i].first.chi2rz+0.07,
+                                                 selectedTracksEmulationWithTruth[j].first.chi2rz+0.07,
+                                                 selectedTracksEmulationWithTruth[i].first.bendchi2+0.07,
+                                                 selectedTracksEmulationWithTruth[j].first.bendchi2+0.07,
+                                                 selectedTracksEmulationWithTruth[i].first.MVA1+0.07,
+                                                 selectedTracksEmulationWithTruth[j].first.MVA1+0.07,
+                                                 vertex.d_T,
+                                                 vertex.R_T,
+                                                 vertex.cos_T,
+                                                 vertex.delta_z*0.125};
+
+      conifer::BDT<ap_fixed<13,8,AP_RND_CONV, AP_SAT>, ap_fixed<13,8,AP_RND_CONV, AP_SAT>> bdt(this->model_);
+      std::vector<ap_fixed<13,8,AP_RND_CONV, AP_SAT>> output = bdt.decision_function(Transformed_features);
+      outputVertex.setScore(output.at(0).to_float());
```

Comment:
Why is this set as a float?

Reply:
It's using the same vertex class and methods as the simulation which expects a float. Also the ntuple maker would have to convert the score to a float anyway before storing in a TBranch, right?